siriuspy/siriuspy/devices/rf.py

```diff
@@ -55,6 +128,418 @@ def frequency(self, value):
+        """."""
+        self.freq_sweep_trig_src = self.SWEEP_TRIG_SRC.AUTO
+        return self._wait('TrigFSweepSrc-Sts', self.SWEEP_TRIG_SRC.AUTO)
+
+    @property
+    def freq_sweep_shape(self):
+        """Define the shape of the sweep.
+
+        Takes the following values:
+            0 - 'SAWT': sawtooth-like waveform:
+                                _           _           _
+                              _| |        _| |        _| |
+                            _|   |      _|   |      _|   |
+                -----|    _|     |    _|     |    _|     |-----
+                     |  _|       |  _|       |  _|
+                     |_|         |_|         |_|
+            1 - 'TRI': triangular waveform:
+                                _                   _
+                              _| |_               _| |_
+                            _|     |_           _|     |_
+                -----|    _|         |_       _|         |_    |-----
+                     |  _|             |_   _|             |_  |
+                     |_|                 |_|                 |_|
+
+        """
+        return self['FreqFreqShp-Sts']
+
+    @freq_sweep_shape.setter
+    def freq_sweep_shape(self, value):
+        self._enum_setter('FreqFreqShp-Sel', value, self.SWEEP_SHAPE)
+
+    @property
+    def freq_sweep_shape_str(self):
+        """Define the shape of the sweep.
+
+        Takes the following values:
+            0 - 'SAWT': sawtooth-like waveform:
+                                _           _           _
+                              _| |        _| |        _| |
+                            _|   |      _|   |      _|   |
+                -----|    _|     |    _|     |    _|     |-----
+                     |  _|       |  _|       |  _|
+                     |_|         |_|         |_|
+            1 - 'TRI': triangular waveform:
+                                _                   _
+                              _| |_               _| |_
+                            _|     |_           _|     |_
+                -----|    _|         |_       _|         |_    |-----
+                     |  _|             |_   _|             |_  |
+                     |_|                 |_|                 |_|
+
+        """
+        return self.SWEEP_SHAPE._fields[self['FreqFreqShp-Sts']]
+
+    def cmd_freq_sweep_shape_to_sawtooth(self):
+        """."""
+        self.freq_sweep_shape = self.SWEEP_SHAPE.SAWT
+        return self._wait('FreqFreqShp-Sts', self.SWEEP_SHAPE.SAWT)
+
+    def cmd_freq_sweep_shape_to_triangular(self):
+        """."""
+        self.freq_sweep_shape = self.SWEEP_SHAPE.TRI
+        return self._wait('FreqFreqShp-Sts', self.SWEEP_SHAPE.TRI)
+
+    @property
+    def freq_sweep_retrace(self):
+        """Define whether to return to start freq. after sweep finish.
+
+        Takes the following values:
+            0 - 'OFF': return to starting frequency.
+            1 - 'ON': stay at stop frequency frequency.
```

Comment:
It seems the text here sould be like in row 393: `1 - 'ON': stay at last frequency of the sweep.`

Reply:
done!